### DISCUSSION TITLE
Connection between kernel's variable and html code cell

### DISCUSSION THREAD

**Discussion by smzm** [microsoft/vscode-jupyter#10647] [New Ideas]
I think it's a great feature if `vscode-jupyter`  has the possibility of the connection between jupyter kernel's variables and `%%html` cells.

It would be great if you can write pure HTML code  (maybe adding some CSS for styling) and put the python kernel variables inside it right away.
Also,  you can set a variable from `HTML` code cell. 

**It helps to create a simple dashboard inside jupyter ipynb files.**

*something like this :* 👇 🎯

- #### Get variable's data from kernel environment
```python
var = 'A variable'
```
```python
%%HTML

<h1 style="color:white"> The result is {{var}} and it's fantastic </h1>
```

- #### Set data into kernel's variables from `HTML` 
```python
%%HTML
You can change it here : 
<label for="{{var}}">Enter a value :</label>
<input type="text" id="{{var}}" name="{{var}}">
```

**Comment by DonJayamanne** [comment]
This would be related to https://github.com/microsoft/vscode-jupyter/issues/4946, though %%html cell magics do not natively support this functionality.

I know this isn't an ideal solution (& likely you are already aware of this option), however thought I'd mention it either way:
```python
# Cell 1
var = 'A variable'
```
```python
# Cell 2
from IPython.display import display, HTML
display(HTML(f'<h1 style="color:white"> The result is {var} and it\'s fantastic </h1>'))
```

**Reply by smzm** [reply]
***Yes, I'm aware of that*** 👍... *but not satisfying*. 

It's good to have kernel variables inside HTML code, but it'll be great if HTML inputs can transmit values to the kernel too.
